src/v3/src/hooks/useOnSubmit.ts

```diff
@@ -47,6 +50,20 @@ export const useOnSubmit = (): (options: OnSubmitHandlerOptions) => Promise<void
       isActionStep,
       stepToRender,
     } = options;
+
+    const handleError = (error: unknown) => {
+      // TODO: handle error based on types
+      // AuthApiError is one of the potential error that can be thrown here
+      // We will want to expose development stage errors from auth-js and file jiras against it
+      setAuthApiError(error as AuthApiError);
+      console.error(error);
+      // error event
+      events?.afterError?.({
+        stepName: currTransaction?.nextStep?.name,
+      }, error);
+      return null;
+    };
```

Comment:
Can you make this a function in the `/util` path, and re-use this here and in widget/index.tsx?

Reply:
yup will do

Reply:
maybe a hook if you want to generalize the logic, since it has component state handling included

Reply:
Ended up not being able to make a hook because im getting the lint error: 
`React Hook "useOnError" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function`.  The custom useOnError hook cant be used in the widget index because it doesnt have access to the context yet.

Reply:
i see, it's a hook already, personally I don't think this handleError fn will be called in other places, since useOnSubmit is  the only centralized place to trigger idx.proceed. For other calls, like idx.start, you probably will want to handle errors differently

Reply:
That same function is used in the `bootstrap` callback and the `resume` callback: https://github.com/okta/okta-signin-widget/blob/0.0/src/v3/src/components/Widget/index.tsx#L173-L202

That is why I recommended extracting to a function, are you ok with this function being duplicated in both places?

Reply:
in this case, it can be wrapped in `useCallback` and shared via context. util dir is used to collect pure functions, so it can be used/tested isolatedly

Reply:
Cool, going to create a ticket for this to follow up and implement this logic